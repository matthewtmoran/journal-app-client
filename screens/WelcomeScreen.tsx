import React from "react";
import { StyleSheet, View, Button } from "react-native";
import { RobotLightText } from "../components/StyledText";
import PrimaryButton from "../components/PrimaryButton";
import SecondaryButton from "../components/SecondaryButton";

const WelcomeScreen = ({ navigation }: any) => {
  return (
    <View style={styles.container}>
      <View>
        <RobotLightText style={styles.message}>
          Welcome To Journal App Î²
        </RobotLightText>
        <PrimaryButton
          styles={styles.button}
          onPress={() => navigation.navigate("Sign In")}
        >
          Log In
        </PrimaryButton>
        <SecondaryButton
          styles={styles.button}
          onPress={() => navigation.navigate("Sign Up")}
        >
          Register
        </SecondaryButton>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    alignContent: "center",
    height: "100%",
  },
  button: {
    marginVertical: 8,
    width: "100%",
  },
  message: {
    fontSize: 30,
    margin: 20,
  },
});

export default WelcomeScreen;
