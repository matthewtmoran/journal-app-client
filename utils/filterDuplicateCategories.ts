import ICategory from "../interfaces/ICategory";

const filterDuplicateCategories = (used: ICategory[], unused: ICategory[]) => {
  const hash: { [key: string]: ICategory } = {};

  for (let i = 0; i < used.length; i++) {
    const item = used[i];
    hash[item.name] = item;
  }

  for (let i = 0; i < unused.length; i++) {
    const item = unused[i];
    hash[item.name] = item;
  }

  return Object.keys(hash).map((key) => {
    return hash[key];
  });
};

export default filterDuplicateCategories;
