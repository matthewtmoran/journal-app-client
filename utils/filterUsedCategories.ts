import ICategory from "../interfaces/ICategory";

const filterUsedCategories = (used: ICategory[], all: ICategory[]) => {
  // if it's in used, remove from all

  const filtered = all.filter((cat) => {
    const exists = used.find((c) => {
      return c.name === cat.name;
    });

    return !exists;
  });

  return filtered ? filtered : [];
};

export default filterUsedCategories;
