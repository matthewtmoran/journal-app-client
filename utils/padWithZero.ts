const padWithZero = (number: number) => {
  const string = number.toString();
  if (number < 10) {
    return "0" + string;
  }
  return string;
};

export default padWithZero;
