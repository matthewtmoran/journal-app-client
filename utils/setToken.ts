import AsyncStorage from "@react-native-community/async-storage";

const setToken = async (token: string) => {
  try {
    await AsyncStorage.setItem("@token", token);
  } catch (e) {
    console.log({ e });
    // saving error
  }
};

export default setToken;
