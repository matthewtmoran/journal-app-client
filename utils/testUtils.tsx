import React from "react";
import { NavigationContainer } from "@react-navigation/native";
import { render } from "@testing-library/react-native";

export function renderWithNavigation(component: any) {
  const App = () => <NavigationContainer>{component}</NavigationContainer>;

  return { ...render(<App />) };
}
