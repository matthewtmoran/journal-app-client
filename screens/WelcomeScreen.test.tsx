import React from "react";
import WelcomeScreen from "./WelcomeScreen";
import { renderWithNavigation } from "../utils/testUtils";
import { fireEvent, render, act } from "@testing-library/react-native";

describe("WelcomeScreen", () => {
  const props = {
    navigation: {
      navigate: jest.fn(),
    },
  };
  it("Should match snapshot", async () => {
    const { baseElement } = render(<WelcomeScreen {...props} />);
    expect(baseElement).toMatchSnapshot();
  });
});
