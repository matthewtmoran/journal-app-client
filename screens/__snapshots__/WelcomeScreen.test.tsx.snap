// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen Should match snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <View>
        <Text
          style={
            Array [
              Object {
                "fontSize": 30,
                "margin": 20,
              },
              Object {
                "fontFamily": "roboto-light",
              },
            ]
          }
        >
          Welcome To Journal App Î²
        </Text>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "backgroundColor": "#144568",
                "borderRadius": 4,
                "marginVertical": 8,
                "paddingHorizontal": 25,
                "paddingVertical": 12,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "roboto-light",
                  },
                ]
              }
            >
              Log In
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "backgroundColor": "#fff",
                "borderRadius": 4,
                "marginVertical": 8,
                "paddingHorizontal": 25,
                "paddingVertical": 12,
                "width": "100%",
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#144568",
                    "fontSize": 18,
                    "textAlign": "center",
                  },
                  Object {
                    "fontFamily": "roboto-light",
                  },
                ]
              }
            >
              Register
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
